src/index.js

```diff
@@ -0,0 +1,82 @@
+/*
+  MIT License http://www.opensource.org/licenses/mit-license.php
+  Author Tobias Koppers @sokra
+*/
+import path from 'path';
+import loaderUtils from 'loader-utils';
+
+export default function fileLoader(content) {
+  this.cacheable && this.cacheable(); // eslint-disable-line no-unused-expressions
+
+  if (!this.emitFile) throw new Error('emitFile is required from module system');
+
+  const query = loaderUtils.getOptions(this) || {};
+  const configKey = query.config || 'fileLoader';
+  const options = this.options[configKey] || {};
+
+  const config = {
+    publicPath: false,
+    useRelativePath: false,
+    name: '[hash].[ext]',
+  };
+
+  // options takes precedence over config
+  Object.keys(options).forEach((attr) => {
+    config[attr] = options[attr];
+  });
+
+  // query takes precedence over config and options
+  Object.keys(query).forEach((attr) => {
+    config[attr] = query[attr];
+  });
+
+  const context = config.context || this.options.context;
+  let url = loaderUtils.interpolateName(this, config.name, {
+    context,
+    content,
+    regExp: config.regExp,
+  });
+
+  let outputPath = '';
+  if (config.outputPath) {
+    // support functions as outputPath to generate them dynamically
+    outputPath = (
+      typeof config.outputPath === 'function' ? config.outputPath(url) : config.outputPath
+    );
+  }
+
+  const filePath = this.resourcePath;
+  if (config.useRelativePath) {
+    const issuerContext = this._module && this._module.issuer // eslint-disable-line no-underscore-dangle
```

Comment:
Maybe we should relax some of these ESLint rules or are these lines subject to change in a smaller upcoming PR by intention ?